runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -520,8 +526,11 @@ public int getSize() {
     private void translateKnownStepCounters(CounterUpdate stepCounterUpdate) {
       CounterStructuredName structuredName =
           stepCounterUpdate.getStructuredNameAndMetadata().getName();
-      if (THROTTLING_MSECS_METRIC_NAME.getNamespace().equals(structuredName.getOriginNamespace())
-          && THROTTLING_MSECS_METRIC_NAME.getName().equals(structuredName.getName())) {
+      if ((THROTTLING_MSECS_METRIC_NAME.getNamespace().equals(structuredName.getOriginNamespace())
+              && THROTTLING_MSECS_METRIC_NAME.getName().equals(structuredName.getName()))
+          || (BIGQUERY_STREAMING_INSERT_THROTTLE_TIME_NAMESPACE.equals(
```

Comment:
Is there a reason why we needed to use a  unique name for BQ but not for GCS or Datastore ?

Reply:
Yes. GCS and Datastore counters are only consumed by batch worker (the `THROTTLING_MSECS_METRIC_NAME` counter here is a separate counter; I am not sure what this is. Maybe all throttling metrics should go to this counter? @ihji ).

Here in the streaming case, precision is on millisecond (whereas GCS and DataStore only store seconds)